### PR TITLE
Tickets/dm 30844

### PR DESCRIPTION
**Pull request by pcortes80**
Please check the code after all changed made. Thanks.

### CODE REVIEW

File: doc/detailsDefault.md [hunk 1]
```diff
@@ -0,0 +1,47 @@
+# Details of Default File
+
+This file stores the **id** of commands, **id** and **compName** of events, and **id** and **compName** of telemetry.
+LabVIEW Component will read this file to register commands, events and telemetry in TCP Server.
+
+The **default.ini** file is in `config/` directory.
+You can follow [here](../config/default.ini) for details.
```
Comment (teweitsai) [inline doc/detailsDefault.md:7]: This (../config/default.ini) is the `ini` file instead of the details.

File: doc/detailsDefault.md [hunk 2]
```diff
@@ -0,0 +1,47 @@
...
+## Explanation of Items
+
+### commands
+
+1. **id** : Specify the name of command that begins with `cmd_`.
+
+### events
+
+1. **compName** : Specify which component's event is being subscribed.
+2. **id** : Specify the name of event that begins with `evt_`.
```
Comment (teweitsai) [inline doc/detailsDefault.md:18]: In the `ini` file, the evt_` is not needed. Right?

File: doc/detailsDefault.md [hunk 3]
```diff
@@ -0,0 +1,47 @@
...
+
+### telemetry
+
+1. **compName** : Specify which component's telemetry is being subscribed.
+2. **id**: Specify the name of telemetry that begins with `tel_`.
```
Comment (teweitsai) [inline doc/detailsDefault.md:23]: In the `ini` file, the `tel_` is not needed. Right?

File: doc/detailsDefault.md [hunk 4]
```diff
@@ -0,0 +1,47 @@
...
+[commands]
+id=move,stop
+```
+
+**For Events/Telemetry:** please write every **compName** with its **id**. Every **compName** and **id** must be separated by a `comma`, for instance, if you want to define the following events:
+
+| compName |          id         |
+|:--------:|:-------------------:|
+|  MTMount | elevationInPosition |
+|   M1M3   |  azimuthInPosition  |
```
Comment (teweitsai) [inline doc/detailsDefault.md:39]: M1M3 has no `azimuthInPosition `. Right?

File: doc/detailsDefault.md [hunk 5]
```diff
@@ -0,0 +1,47 @@
...
+
+Then, write them in this way:
+
+```text
+[events]
+compName=MTMount,M1M3
```
Comment (teweitsai) [inline doc/detailsDefault.md:45]: M1M3 has no `azimuthInPosition `. Right?

File: simulator/README.md [hunk 6]
```diff
@@ -0,0 +1,97 @@
...
+
+TCP Server will receive Events that LabVIEW component subscribed through the [default.ini](../config/default.ini) file.
+
+### Stop Component Simulator
+
+To stop the Component Simulator click to "Stop Simulator" button.
+
+Note:
+
+1. You can only stop the Simulator Component if you stop de TCP Server before.
```
Comment (teweitsai) [inline simulator/README.md:96]: if you stop de TCP Server before --> if you stop "the" TCP Server before.

File: doc/detailsDefault.md [hunk 7]
```diff
@@ -0,0 +1,47 @@
...
+LabVIEW Component will read this file to register commands, events and telemetry in TCP Server.
+
+The **default.ini** file is in `config/` directory.
+You can follow [here](../config/default.ini) for details.
+
+## Explanation of Items
+
+### commands
+
+1. **id** : Specify the name of command that begins with `cmd_`.
```
[the remark this answers is not in the document]
Comment (teweitsai) [reply doc/detailsDefault.md:13]: In addition, I felt the `name` is better than `id` here. But I do understand why you pick the `id` here.

**Review by teweitsai — APPROVED** [review]
This update looks good to me. I left some inline comments of documents. For the code, the following is my review:

[testCreateConfigFilePath.vi]

1. Why do you need the "Clear Errors.vi" in the block diagram? Does it fail in the CTIO Jenkins?

[readConfigFileDefault.vi]

1. Based on your code design, we could get ride of readConfigCmd.vi, openConfigFile.vi, and closeConfigFile.vi. And we can combine the readConfigEvi.vi and readConfigTel.vi to be someething like the readConfigEvtTel.vi and use an enum to decide the event or telemetry.

[openConfigFile.vi]

1. I think this vi can reuse the createConfigFilePath.vi. Or just get ride of this vi.

[readConfigEvt.vi]

1. You can combine this with readConfigTel.vi.

[readConfigTel.vi]

1. 1. You can combine this with readConfigEvt.vi.

File: doc/detailsDefault.md (hunk 1 above)
Comment (pcortes80) [reply doc/detailsDefault.md:7]: the idea is click and go to the ini file, so I just modified the text.

File: doc/detailsDefault.md (hunk 7 above)
[the remark this answers is not in the document]
Comment (pcortes80) [reply doc/detailsDefault.md:13]: You are right, I will remove  the `cmd_`.  I picked **id** to be consistence with `README.md` file.

File: doc/detailsDefault.md (hunk 2 above)
Comment (pcortes80) [reply doc/detailsDefault.md:18]: You are right, I will remove the `evt_`.

File: doc/detailsDefault.md (hunk 3 above)
Comment (pcortes80) [reply doc/detailsDefault.md:23]: You are right, I will remove the `tel_ `.

File: doc/detailsDefault.md (hunk 4 above)
Comment (pcortes80) [reply doc/detailsDefault.md:39]: Modified M1M3 by MTMount.

File: doc/detailsDefault.md (hunk 5 above)
Comment (pcortes80) [reply doc/detailsDefault.md:45]: Modified M1M3 by MTMount.

File: simulator/README.md (hunk 6 above)
Comment (pcortes80) [reply simulator/README.md:96]: Fixed. Thanks.

**Comment by pcortes80** [comment]
Hi, these are my replies:

[testCreateConfigFilePath.vi]

1. Why do you need the "Clear Errors.vi" in the block diagram? Does it fail in the CTIO Jenkins?

<Reply> This function is not necessary; I just removed this.

[readConfigFileDefault.vi]

1. Based on your code design, we could get ride of readConfigCmd.vi, openConfigFile.vi, and closeConfigFile.vi. And we can combine the readConfigEvi.vi and readConfigTel.vi to be someething like the readConfigEvtTel.vi and use an enum to decide the event or telemetry.

<Reply> 
- removed readConfigCmd.vi.
- removed closeConfigFile.vi.
- removed openConfigFile.vi.
- combined readConfigEvt.vi and readConfigTel.vi into readConfigEvtTel.vi

[openConfigFile.vi]

1. I think this vi can reuse the createConfigFilePath.vi. Or just get ride of this vi.

<Reply> Removed openConfigFile.vi.

[readConfigEvt.vi]

1. You can combine this with readConfigTel.vi.

<Reply> Done.

[readConfigTel.vi]

1. You can combine this with readConfigEvt.vi.

<Reply> Done.

**Pull request merged**